src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,59 @@
 package core.stack;
 
+import java.util.Arrays;
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int RESIZE_ELEMENT_DATA = 2;
+    private Object[] elementData;
+    private int size;
+
+    public MateStack() {
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
 
+    public void push(T value) {
+        if (size == elementData.length) {
+            resize();
+        }
+        elementData[size++] = value;
     }
 
     public T peek() {
-        return null;
+        getCheckSize();
+        return (T) elementData[size - 1];
     }
 
     public T pop() {
-        return null;
+        T value = peek();
+        removeElement(size() - 1);
+        return value;
     }
 
     public int size() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        elementData = Arrays.copyOf(elementData,
+                DEFAULT_CAPACITY * RESIZE_ELEMENT_DATA);
+    }
+
+    private void removeElement(int index) {
+        if (index >= size || index < 0) {
+            throw new ArrayIndexOutOfBoundsException("Index incorrect");
+        }
```

Comment:
are you sure that you need this check? you will pass `size - 1` as index to this method